PharmacyManagementAPI/Program.cs

```diff
@@ -6,16 +6,27 @@
 using PharmacyManagementAPI.Repositories; 
 using PharmacyManagementAPI.Services;
 using System.Text.Json;
+using PharmacyManagementAPI;
+using System.Text.Json.Serialization;
+using PharmacyManagementAPI.Filters;
 
 var builder = WebApplication.CreateBuilder(args);
+// Add AutoMapper
+builder.Services.AddAutoMapper(typeof(MappingProfile));
 
 //Add services to the container.
-
 builder.Services.AddControllers();
```

Comment:
Add these lines to ensure that your HTTP request pipeline only handles properly formatted JSON input

builder.Services.AddControllers(options  => 
{
    options.Filters.Add(new ProducesAttribute("application/json"));
    options.Filters.Add(new ConsumesAttribute("application/json"));
});